src/main/java/com/aws/greengrass/clientdevices/auth/CertificateManager.java

```diff
@@ -218,12 +218,10 @@ private void subscribeToServerCertificateUpdatesNoCSR(@NonNull GetCertificateReq
         cisShadowMonitor.addToMonitor(certificateGenerator);
         caConfigurationMonitor.addToMonitor(certificateGenerator);
 
-        // TODO: Doing this here is wrong. We are assuming that we can call this and it would work but this should
-        //  only be called after we are sure a certificate authority has been configured. Which might not be true
-        //  before this gets called. Having this fail several times in a row (if no CA has been configured) will cause
-        //  downstream components that rely on this to fail
-        certificateGenerator.generateCertificate(connectivityInformation::getCachedHostAddresses,
-                "initialization of server cert subscription");
+        if (certificateStore.isReady()) {
```

Comment:
this call is happening when creating the subscription, so when will the cert get generated now that this code may not execute?


Also, if this code will not execute sometimes, why not remove it completely and let the cert get generated by whatever workflow is fixing it in the case when `isReady()` is false?

Reply:
If customers use a config to provide their own certificate and launch the component for the first time. If they provide a wrong path for the cert in the config we fail to load it and no CA is configured. Still, components will call this method which will in turn raise an exception because there is no CA, causing them to go eventually into a broken state.

To your point, we could completely remove the conditionals once we refactor all this subscription stuff to work reacting to events.

Reply:
So you described the current state I believe. But what happens with this change, now that generate certificates isn't called?

How do the certs eventually get generated?

Reply:
When we set the ca chain with the pk, we emit an event. We have a handler that listens to it and triggers rotation every time a new CA is configured